nuxeo-runtime/nuxeo-runtime-test/src/main/java/org/nuxeo/runtime/test/NXRuntimeTestCase.java

```diff
@@ -270,74 +268,20 @@ protected OSGiRuntimeService handleNewRuntime(OSGiRuntimeService aRuntime) {
     }
 
     public static URL[] introspectClasspath(ClassLoader loader) {
-        // normal case
-        if (loader instanceof URLClassLoader) {
-            return ((URLClassLoader) loader).getURLs();
-        }
-        // surefire suite runner
-        final Class<? extends ClassLoader> loaderClass = loader.getClass();
-        if (loaderClass.getName().equals("org.apache.tools.ant.AntClassLoader")) {
-            try {
-                Method method = loaderClass.getMethod("getClasspath");
-                String cp = (String) method.invoke(loader);
-                String[] paths = cp.split(File.pathSeparator);
-                URL[] urls = new URL[paths.length];
-                for (int i = 0; i < paths.length; i++) {
-                    urls[i] = new URL("file:" + paths[i]);
-                }
-                return urls;
-            } catch (NoSuchMethodException | SecurityException | IllegalAccessException | IllegalArgumentException
-                    | InvocationTargetException | MalformedURLException cause) {
-                throw new AssertionError("Cannot introspect mavent class loader", cause);
-            }
-        }
-        // try getURLs method
-        try {
-            Method m = loaderClass.getMethod("getURLs");
-            return (URL[]) m.invoke(loader);
-        } catch (NoSuchMethodException | SecurityException | IllegalAccessException | IllegalArgumentException
-                | InvocationTargetException cause) {
-            throw new AssertionError("Unsupported classloader type: " + loaderClass.getName()
-                    + "\nWon't be able to load OSGI bundles");
-        }
-    }
+        return new FastClasspathScanner().getUniqueClasspathElements().stream()
+                .map(file -> {
+                    try {
+                        return file.toURI().toURL();
+                    } catch (MalformedURLException cause) {
+                        throw new Error("Could not get URL from " + file, cause);
```

Comment:
`Error` is for low-level system exception. Please use `RuntimeException` in cases like this.